src/xform_to_datalog/authorization_logic_test.cc

```diff
@@ -0,0 +1,80 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+#include "src/xform_to_datalog/authorization_logic.h"
+
+#include <cstdlib>
+#include <filesystem>
+#include <fstream>
+#include <vector>
+
+#include "src/common/testing/gtest.h"
+
+namespace fs = std::filesystem;
+
+#include <iostream>
+
+namespace raksha::xform_to_datalog {
+
+class AuthorizationLogicTest : public ::testing::Test {
+public:
+  static fs::path GetTestDataDir() {
+    const char* test_srcdir_env = std::getenv("TEST_SRCDIR");
+    const char* test_workspace_env = std::getenv("TEST_WORKSPACE");
```

Comment:
Doesn't this use of environment variables make this test non-hermetic?

Reply:
These are set by `bazel` when we invoke `bazel test`. See https://docs.bazel.build/versions/main/test-encyclopedia.html#initial-conditions

Reply:
These are set by `bazel` when we invoke `bazel test`. See https://docs.bazel.build/versions/main/test-encyclopedia.html#initial-conditions